src/puffin/src/puffin_manager/stager/bounded_stager.rs

```diff
@@ -284,6 +306,15 @@ impl Stager for BoundedStager {
 
         Ok(())
     }
+
+    async fn purge(&self, puffin_file_name: &str) -> Result<()> {
+        let keys = self.file_mapping.take_keys(puffin_file_name).await;
+        for key in keys {
+            let cache_key = Self::encode_cache_key(puffin_file_name, &key);
+            self.cache.invalidate(&cache_key).await;
+        }
+        Ok(())
+    }
```

Comment:
Maybe we can use `invalidate_entries_if` so we don't have to maintain the mapping.
https://docs.rs/moka/latest/moka/future/struct.Cache.html#method.invalidate_entries_if